mods/taskbar-on-top.wh.cpp

```diff
@@ -1245,7 +1275,19 @@ HRESULT WINAPI DwmSetWindowAttribute_Hook(HWND hwnd,
     int cy = targetRect.bottom - targetRect.top;
 
     if (target == Target::StartMenu) {
-        // Only change height.
+        // Obter o monitor do hwnd
+        HMONITOR monitor = MonitorFromWindow(hwnd, MONITOR_DEFAULTTONEAREST);
+
+        MONITORINFO monitorInfo = {sizeof(MONITORINFO)};
+        if (!GetMonitorInfo(monitor, &monitorInfo)) {
+            return original(); // Fallback caso falhe.
+        }
+
+        // Centralizar no monitor ativo
+        int xNew = (monitorInfo.rcWork.left + monitorInfo.rcWork.right - cx) / 2;
+        int yNew = monitorInfo.rcWork.top;
```

Comment:
This seems to be the real fix. It's not clear to me why x and y have to be adjusted. I'd expect them to be handled by the taskbar and be correct already. Perhaps the issue is caused by something else, like another mod. How can I reproduce it? Does it occur if all other mods are disabled?

Reply:
Yes. This behavior occur if I enable just this mod. 
My windows 11 version is 24h2 build 26100.2605. 

I noticed anothers strange behaviors:
- Even though adjust is applied when you click in Power or in profile image the startup menu is displaced to right side of screen.
- The startup menu slides from mid of screen to top. 
- The notifications slides from bottom right side not from top right side following taskbar position.

Reply:
> Yes. This behavior occur if I enable just this mod.
> My windows 11 version is 24h2 build 26100.2605.

I couldn't reproduce the problem on my Win11 23H2. I don't have a 24H2 computer with multiple monitors right now, I'll try to reproduce it later. I prefer to understand what's going on there and why this check is needed.

Meanwhile, can you try the original code with this line removed: `*wParam = ABE_TOP;`? Does it fix the misplaced start menu?

Also, can you post a video recording of how it looks, as well as debug logs? Perhaps it will help reproduce the problem.

Please attach debug logs by following the instructions on the Troubleshooting page:
https://github.com/ramensoftware/windhawk/wiki/Troubleshooting#some-or-all-windhawk-mods-dont-work

> * Disable the mod.
> * Go to the mod's Advanced tab.
> * Set **Debug logging** to **Detailed debug logs**.
> * Click on **Show log output**.
> * Enable the mod.
> * Verify that the mod doesn't work, copy the log and save it to a file.
> * Set **Debug logging** back to **None**.

<hr>

> * Even though adjust is applied when you click in Power or in profile image the startup menu is displaced to right side of screen.

That's another reason to prefer getting to the root cause.

> * The startup menu slides from mid of screen to top.
> * The notifications slides from bottom right side not from top right side following taskbar position.

These are known limitations, see https://github.com/ramensoftware/windhawk-mods/issues/1157.